public/editor/scripts/editor/js/fc/bramble-ui-bridge.js

```diff
@@ -443,6 +443,16 @@ define(function(require) {
       }
     });
 
+    bramble.on("projectDirty", function(){
+      $("#navbar-dirt-file").addClass("hide");
```

Comment:
You need to do this only for the "not-signed-in" case. I bet that can be done with a `!Project.getUser()` check